addons/folo/common/src/main/java/org/commonjava/indy/folo/data/DtxTrackingRecord.java

```diff
@@ -0,0 +1,271 @@
+package org.commonjava.indy.folo.data;
+
+
+import com.datastax.driver.core.Row;
+import com.datastax.driver.mapping.annotations.Column;
+import com.datastax.driver.mapping.annotations.PartitionKey;
+import com.datastax.driver.mapping.annotations.Table;
+import org.commonjava.indy.folo.model.StoreEffect;
+import org.commonjava.indy.folo.model.TrackedContentEntry;
+import org.commonjava.indy.folo.model.TrackingKey;
+import org.commonjava.indy.model.core.AccessChannel;
+import org.commonjava.indy.model.core.StoreKey;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Table(name = "records" )
+public class DtxTrackingRecord {
+
+    private final static Boolean SEALED = true;
+    private final static Boolean IN_PROGRESS = false;
+
+
+    @PartitionKey(0)
+    @Column(name = "tracking_key")
+    String trackingKey;
+
+    @Column(name = "sealed")
+    Boolean state;
+
+    @Column(name = "store_key")
+    String storeKey;
+
+    @Column(name = "access_channel")
+    String  accessChannel;
+
+    @PartitionKey(1)
+    @Column(name = "path")
+    String  path;
+
+    @Column(name = "origin_url")
+    String  originUrl;
+
+    @Column(name = "local_url")
+    String localUrl;
+
+    @Column(name = "store_effect")
+    String storeEffect;
+
+    @Column(name = "md5")
+    String  md5;
+
+    @Column(name = "sha256")
+    String sha256;
+
+    @Column(name = "sha1")
+    String sha1;
+
+    @Column(name = "size")
+    Long size;
+
+    @Column(name = "timestamps")
+    Set<Long> timestamps;
```

Comment:
Ugh, I missed this the first time. In the C* implementation, we should make this a single timestamp, instead of a set of timestamps. The set form is used to capture multiple accesses to the same path in a build, and it only gets used when we create the full tracking record for the build...when we collect all the path records in to a sealed record, we were de-duplicating the path records and accumulating the access timestamps. However, in this implementation we're only storing the path-level records (since it doesn't make sense to collect the sealed record when we have an efficient way to query for all path-level records.

Reply:
Yes, you are right. I was thinking that is representation. of first field entry (in-progress ) and sealing record(sealed).
Should i leave this set of timestamps and add another field for entry and finish?